File: JavaScript_Courses/JavaScript_Basics/Preparation/02_Conditional_Statements_Exercise/03_Time_Plus_15_Minutes.js
function timePlus15Minutes(input) {
    let hour = Number(input[0]);
    let minutes = Number(input[1]);

    let newHour = hour;
    let newMinutes = minutes + 15;
    if (newMinutes >= 60) {
        newMinutes -= 60;
        newHour += 1;
        if (newHour == 24) {
            newHour = 0;
        }
    }

    if (newMinutes < 10) {
        console.log(`${newHour}:0${newMinutes}`);
    } else {
        console.log(`${newHour}:${newMinutes}`);
    }
}

// timePlus15Minutes(["1", "46"]);
// timePlus15Minutes(["23", "59"]);

File: Python_Courses/Python_Basics/Python_PB_2024_10/02_Conditional_Statements_Exercise/03_Time_plus_15_Minutes.py
hours = int(input())
minutes = int(input())

minutes_plus_15 = minutes + 15

if minutes_plus_15 >= 60:  # 65 -> 5; +1h
    minutes_plus_15 -= 60  # minutes_plus_15 = minutes_plus_15 - 60
    hours += 1  # hours = hours + 1

if hours == 24:
    hours = 0

print(f'{hours}:{minutes_plus_15 :02d}')
